specification/maintenance/resource-manager/Microsoft.Maintenance/preview/2020-07-01-preview/Maintenance.json

```diff
@@ -0,0 +1,1669 @@
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ListConfigurationAssignmentsResult": {
+      "description": "Response for ConfigurationAssignments list",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of configuration Assignments",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConfigurationAssignment"
+          }
+        }
+      }
+    },
+    "MaintenanceWindow": {
+      "description": "Definition of a MaintenanceWindow",
+      "type": "object",
+      "properties": {
+        "startDateTime": {
+          "description": "Effective start date of the maintenance window in YYYY-MM-DD hh:mm:ss format. The start date can be set to either the current date or future date. The window will be created in the time zone provided and adjusted to daylight savings according to that time zone.",
+          "type": "string"
+        },
+        "expirationDateTime": {
+          "description": "Effective expiration date of the maintenance window in YYYY-MM-DD hh:mm:ss format. The window will be created in the time zone provided and adjusted to daylight savings according to that time zone. Expiration date must be set to a future date. If not provided, it will be set to the maximum datetime 9999-12-31 23:59:59.",
+          "type": "string"
+        },
+        "duration": {
+          "description": "Duration of the maintenance window in HH:mm format. If not provided, default value will be used based on maintenance scope provided. Example: 05:00.",
+          "type": "string"
+        },
+        "timeZone": {
+          "description": "Name of the timezone. List of timezones can be obtained by executing [System.TimeZoneInfo]::GetSystemTimeZones() in PowerShell. Example: Pacific Standard Time, UTC, W. Europe Standard Time, Korea Standard Time, Cen. Australia Standard Time.",
+          "type": "string"
+        },
+        "recurEvery": {
+          "description": "Rate at which a Maintenance window is expected to recur. The rate can be expressed as daily, weekly, or monthly schedules. Daily schedule are formatted as recurEvery: [Frequency as integer]['Day(s)']. If no frequency is provided, the default frequency is 1. Daily schedule examples are recurEvery: Day, recurEvery: 3Days.  Weekly schedule are formatted as recurEvery: [Frequency as integer]['Week(s)'] [Optional comma separated list of weekdays Monday-Sunday]. Weekly schedule examples are recurEvery: 3Weeks, recurEvery: Week Saturday,Sunday. Monthly schedules are formatted as [Frequency as integer]['Month(s)'] [Comma separated list of month days] or [Frequency as integer]['Month(s)'] [Week of Month (First, Second, Third, Fourth, Last)] [Weekday Monday-Sunday]. Monthly schedule examples are recurEvery: Month, recurEvery: 2Months, recurEvery: Month day23,day24, recurEvery: Month Last Sunday, recurEvery: Month Fourth Monday.",
+          "type": "string"
+        }
```

Comment:
Looks like all the properties are optional. Do you have default values if these are not provided.

Reply:
A maintenance window is optional. If none of these values are provided then we will not create a maintenance window with the maintenance config for the customer. If some window properties are set, then we will check for startTime,TimeZone,recurEvery to be set as they are required. Duration/expirationdatetime are optional and I have specified their defaults in the descriptions.